app/helpers/products_helper.rb

```diff
@@ -2,4 +2,16 @@ module ProductsHelper
   def format_html_to_overview(html, **truncate_args)
     truncate(strip_tags(html), truncate_args)
   end
+
+  def generate_ratings
+    (0.0..5.0).step(0.5).to_a
+  end
+
+  def average_rating(product)
+    product.ratings.average(:rating) || 0
```

Comment:
why display 0 if no one has rated before? show message like - Be the First to rate